pandas/core/window/rolling.py

```diff
@@ -460,15 +478,21 @@ def hfunc(values: ArrayLike) -> ArrayLike:
             # GH#42736 operate column-wise instead of block-wise
             try:
                 res = hfunc(arr)
-            except (TypeError, NotImplementedError):
+            except TypeError:
                 pass
+            except NotImplementedError as err:
```

Comment:
If `step` validation can be done in `BaseWindow._validate` or the appropriate `__init__` this can be avoided